src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECNamedCurve.cs

```diff
@@ -0,0 +1,454 @@
+        public static ECNamedCurve secP384r1
+        {
+            get
+            {
+                return s_secP384r1 ?? (s_secP384r1 = CreateCurve(nameof(secP384r1)));
+            }
+        }
+
+        private static ECNamedCurve s_secP521r1;
+        public static ECNamedCurve secP521r1
+        {
+            get
+            {
+                return s_secP521r1 ?? (s_secP521r1 = CreateCurve(nameof(secP521r1)));
+            }
+        }
+
+        private static ECNamedCurve s_wtls12;
+        public static ECNamedCurve wtls12
+        {
+            get
+            {
+                return s_wtls12 ?? (s_wtls12 = CreateCurve(nameof(wtls12)));
+            }
+        }
+
+        private static ECNamedCurve s_wtls7;
+        public static ECNamedCurve wtls7
+        {
+            get
+            {
+                return s_wtls7 ?? (s_wtls7 = CreateCurve(nameof(wtls7)));
+            }
+        }
+
+        private static ECNamedCurve s_wtls9;
+        public static ECNamedCurve wtls9
+        {
+            get
+            {
+                return s_wtls9 ?? (s_wtls9 = CreateCurve(nameof(wtls9)));
+            }
+        }
+
+        private static ECNamedCurve s_x962P192v1;
+        public static ECNamedCurve x962P192v1
+        {
+            get
+            {
+                return s_x962P192v1 ?? (s_x962P192v1 = CreateCurve(nameof(x962P192v1)));
+            }
+        }
+
+        private static ECNamedCurve s_x962P192v2;
+        public static ECNamedCurve x962P192v2
+        {
+            get
+            {
+                return s_x962P192v2 ?? (s_x962P192v2 = CreateCurve(nameof(x962P192v2)));
+            }
+        }
+
+        private static ECNamedCurve s_x962P192v3;
+        public static ECNamedCurve x962P192v3
+        {
+            get
+            {
+                return s_x962P192v3 ?? (s_x962P192v3 = CreateCurve(nameof(x962P192v3)));
+            }
+        }
+
+        private static ECNamedCurve s_x962P239v1;
+        public static ECNamedCurve x962P239v1
+        {
+            get
+            {
+                return s_x962P239v1 ?? (s_x962P239v1 = CreateCurve(nameof(x962P239v1)));
+            }
+        }
+
+        private static ECNamedCurve s_x962P239v2;
+        public static ECNamedCurve x962P239v2
+        {
+            get
+            {
+                return s_x962P239v2 ?? (s_x962P239v2 = CreateCurve(nameof(x962P239v2)));
+            }
+        }
+
+        private static ECNamedCurve s_x962P239v3;
+        public static ECNamedCurve x962P239v3
+        {
+            get
+            {
+                return s_x962P239v3 ?? (s_x962P239v3 = CreateCurve(nameof(x962P239v3)));
+            }
+        }
+
+        private static ECNamedCurve s_x962P256v1;
+        public static ECNamedCurve x962P256v1
+        {
+            get
+            {
+                return s_x962P256v1 ?? (s_x962P256v1 = CreateCurve(nameof(x962P256v1)));
+            }
+        }
```

Comment:
Do we care whether a race to access these properties will result in multiple instances?

You could also cut down on the boilerplate in these properties using the C# 6 expression-body member syntax.